### ISSUE TITLE
Don't log out on failure to authenticate with homeserver

### ISSUE THREAD

**Issue by RyanGibb** [vector-im/element-web#22728]
#### What would you like to do?

Prevent the client from logging out when failing to authenticate.

#### Why would you like to do it?

I recently updated my server machine, which changed the postgres server from version 13 to 14. This created a new database which, removing my user account. On failing to authenticate, all my element clients logged out. When restoring the postgres v13 database, I had to reauthenticate all my clients. Luckily one was offline for the duration of this disaster, so I could recover the encryption keys without the security key. But it would be nice if the element client retained data like encryption keys until the user explicitly delete it.

**Comment by bwindels** [comment]
Fwiw, the common solution to this problem is using key backup.

**Comment by RyanGibb** [comment]
> Fwiw, the common solution to this problem is using key backup.

Unfortunately I was negligent enough to lose the security key 😄 But I take the point.

**Comment by pedro-nonfree** [comment]
There is a very weird situation that I was describing in #22859 , after some *fill disk* state, a user with a running session (that still works) cannot perform authenticated actions (such as removing the other devices or reset the device) because the auth is not working. New logins are not allowed. That looks like a deadlock. Any suggestion on how to solve it?

**Comment by t3chguy** [comment]
@pedro-nonfree those actions are blocked by your server failing to action the request, I suggest asking for workarounds in the support of whichever server implementation you are connecting to.

**Comment by pedro-nonfree** [comment]
yes @t3chguy, I agree on what you say. Sorry, I missed to update this issue too, we are asking for workarounds on the server implementation part here ( https://github.com/matrix-org/synapse/issues/13305 )
